flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/JobReconciler.java

```diff
@@ -82,7 +83,11 @@ public void reconcile(FlinkDeployment flinkApp, Context context, Configuration e
         }
 
         boolean specChanged = !flinkApp.getSpec().equals(lastReconciledSpec);
-        if (specChanged && readyForSpecChanges(flinkApp)) {
+        if (specChanged) {
+            if (!readyForSpecChanges(flinkApp)) {
```

Comment:
Could we please rename this method and changed the logged line? `readyForSpecChanges` is a little confusing.

We could call it `inUpgradeableState` that would describe the logic better